src/autoscaler/metricscollector/cmd/metricscollector/metricscollector_test.go

```diff
@@ -105,6 +105,8 @@ var _ = Describe("MetricsCollector", func() {
 
 			It("returns with a 200", func() {
 				rsp, err := http.Get(fmt.Sprintf("http://127.0.0.1:%d/v1/apps/an-app-id/metrics/memory", mcPort))
+				defer rsp.Body.Close()
```

Comment:
I saw you changed to 
```
				rsp, err := http.Get(fmt.Sprintf("http://127.0.0.1:%d/v1/apps/an-app-id/metrics/memory", mcPort))

				Expect(err).NotTo(HaveOccurred())
				Expect(rsp.StatusCode).To(Equal(http.StatusOK))
				rsp.Body.Close()
```
If Expect(rsp.StatusCode).To(Equal(http.StatusOK)) fails the test, then rsp.Body.Close() will be skipped.  So should we do like following?
```
				rsp, err := http.Get(fmt.Sprintf("http://127.0.0.1:%d/v1/apps/an-app-id/metrics/memory", mcPort))
				Expect(err).NotTo(HaveOccurred())
                                defer rsp.Body.Close()
				Expect(rsp.StatusCode).To(Equal(http.StatusOK))
```